### ISSUE TITLE
Incorrect artity calculation for optional block arguments

### ISSUE THREAD

**Issue by makmic** [makandra/consul#47]
While consul version 1.0.1 fixed the arity of powers taking block arguments, it led to a bug where the arity is calculated incorretly for optional arguments. 

## An example
Given this power:
```
power :fruits do |season, temperature = nil|
  if season == :summer
     Fruit.all
  elsif temperature
    ...
  end
end
```
In consul 1.0, it's arity was -1, now it claims to be 0:
```
subject.method(:fruits).arity # 0
```
The effect of this is that I canot call this power with the right number of arguments:

```
subject.fruits(:summer)
# => *** ArgumentError Exception: wrong number of arguments (given 1, expected 0)

subject.fruits(:summer, 24)
# => *** ArgumentError Exception: wrong number of arguments (given 2, expected 0)

subject.fruits()
# => *** ArgumentError Exception: wrong number of arguments (given 0, expected 1..2)
```

**Comment by triskweline** [comment]
This needs to be fixed in https://github.com/makandra/memoized .

I think this can be fixed for methods with optional arguments that default to `nil`. I'm not sure it can be fixed for optional arguments that default to another value. While we can detect if a parameter is optional (using `#arity` or `#parameters`), there is currently [no way to retrieve a parameter's default value](https://bugs.ruby-lang.org/issues/8629) in Ruby.

**Comment by makmic** [comment]
I didn't expect the problem to be so deeply rooted in the Ruby language. The affected powers of the project where I found this issue can be easily transformed in a non-optional style, so I am not restrained to an earlier version of consul.

Please change the settings of the `memoized` GitHub project to allow the creation of issues.Thank you for looking into this!

**Comment by triskweline** [comment]
There might be a way for `memoized` to support this fully. I need to look into it.

memoized now accepts issues.

**Comment by triskweline** [comment]
Fixed in 1.0.2.

**Comment by makmic** [comment]
Thank you! It works like a charm :slightly_frowning_face:
